style.md

```diff
@@ -1894,6 +1896,167 @@ this contract. Specifying field names inside tags makes the contract explicit,
 and it guards against accidentally breaking the contract by refactoring or
 renaming fields.
 
+### Don't fire-and-forget goroutines
+
+Goroutines are lightweight, but they're not free.
+Do not leak goroutines in production code.
+In general, every goroutine:
+
+- must have a predictable time at which it will stop running; or
+- there must be a way to signal to the goroutine that it should stop
+
+In both cases, there must be a way code to block and wait for the goroutine to
+finish.
+
+For example:
+
+<table>
+<thead><tr><th>Bad</th><th>Good</th></tr></thead>
+<tbody>
+<tr><td>
+
+```go
+go func() {
+  for {
+    flush()
+    time.Sleep(delay)
+  }
+}()
+```
+
+</td><td>
+
+```go
+stop := make(chan struct{})
+go func() {
+  ticker := time.NewTicker(delay)
+  defer ticker.Stop()
+  for {
+    select {
+    case <-tick.C:
+      flush()
+    case <-stop:
+      return
+    }
+  }
+}()
+```
+
+</td></tr>
+<tr><td>
+
+There's no way to stop this goroutine.
+This will run until the application exits.
+
+</td><td>
+
+This goroutine can be stopped with `close(stop)`.
+
+</td></tr>
+</tbody></table>
+
+Given a means of stopping the goroutine,
+there must be a way to wait for it to stop.
+There are two popular ways to do this:
+
+- Use a `sync.WaitGroup`.
+  Do this if there are multiple goroutines that you want to wait for
+
+    ```go
+    var wg sync.WaitGroup
+    wg.Add(N)
+    for i := 0; i < N; i++ {
+      go func() {
+        defer wg.Done()
+        // ...
+      }()
+    }
+    
+    // To wait for all to finish:
+    wg.Wait()
+    ```
+
+- Add another `chan struct{}` that the goroutine closes when it's done.
+  Do this if there's only one goroutine.
+
+    ```go
+    done := make(chan struct{})
+    go func() {
+      defer close(done)
+      // ...
+    }()
+    
+    // To wait for the goroutine to finish:
+    <-done
+    ```
+
+#### No goroutines in `init()`
+
+`init()` functions should almost never spawn goroutines.
```

Comment:
it says "almost never", but doesn't give an example of when it may be possible (within an `init()` call)

Reply:
I'll drop the "almost never" until we can come up with reasonable exceptions.